NodeMCU v1.0 board option "Builtin Led" not working

### Basic Infos

- [X] This issue complies with the [issue POLICY doc](https://github.com/esp8266/Arduino/blob/master/POLICY.md).
- [X] I have read the documentation at [readthedocs](https://arduino-esp8266.readthedocs.io/en/latest) and the issue is not addressed there.
- [X] I have tested that the issue is present in current master branch (aka latest git).
- [X] I have searched the issue tracker for a similar issue.
- [X] If there is a stack dump, I have decoded it.
- [X] I have filled out all fields below.

#### Platform

- Hardware:					ESP-12E
- Core Version:			2.6.1
- Development Env:		Sloeber (as an Eclipse plugin)
- Operating System:		Devuan

### Settings in IDE

- Module:  			NodeMcu 1.0 (ESP-12E Module)
- Flash Mode:			don't konw
- Flash Size: 		4MB
- lwip Variant:   v2 Lower Memory
- Reset Method:		don't know
- Flash Frequency:	don't know
- CPU Frequency:		160MHz
- Upload Using:		SERIAL
- Upload Speed:   460800


### Problem Description

NodeMCU v1.0 board configuration option "Builtin Led" (since v2.6.1) has no effect because `LED_BUILTIN` is overwritten by `variants/nodemcu/pins_arduino.h`. A compiler warning also thrown (more than once).
```
...
Building file: /home/erdem/eclipse/ide-latest-released/eclipse/arduinoPlugin/packages/esp8266/hardware/esp8266/2.6.1/cores/esp8266/uart.cpp
Starting C++ compile
"/home/erdem/eclipse/ide-latest-released/eclipse//arduinoPlugin/packages/esp8266/tools/xtensa-lx106-elf-gcc/2.5.0-4-b40a506/bin/xtensa-lx106-elf-g++" -D__ets__ -DICACHE_FLASH -U__STRICT_ANSI__ "-I/home/erdem/eclipse/ide-latest-released/eclipse//arduinoPlugin/packages/esp8266/hardware/esp8266/2.6.1/tools/sdk/include" "-I/home/erdem/eclipse/ide-latest-released/eclipse//arduinoPlugin/packages/esp8266/hardware/esp8266/2.6.1/tools/sdk/lwip2/include" "-I/home/erdem/eclipse/ide-latest-released/eclipse//arduinoPlugin/packages/esp8266/hardware/esp8266/2.6.1/tools/sdk/libc/xtensa-lx106-elf/include" "-I/home/erdem/workspace/priv/Arduino/SandboxESP8266/Release/core" -c -Wall -Wextra -Os -g -mlongcalls -mtext-section-literals -fno-rtti -falign-functions=4 -std=gnu++11 -MMD -ffunction-sections -fdata-sections -fno-exceptions -DNEW_OOM_ABORT  -DNONOSDK22x_191024=1 -DF_CPU=160000000L -DLWIP_OPEN_SRC -DTCP_MSS=536 -DLWIP_FEATURES=1 -DLWIP_IPV6=0   -DARDUINO=10802 -DARDUINO_ESP8266_NODEMCU -DARDUINO_ARCH_ESP8266 -DARDUINO_BOARD="\"ESP8266_NODEMCU\"" -DLED_BUILTIN=16 -DFLASHMODE_DIO  -DESP8266   -I"/home/erdem/eclipse/ide-latest-released/eclipse/arduinoPlugin/packages/esp8266/hardware/esp8266/2.6.1/cores/esp8266" -I"/home/erdem/eclipse/ide-latest-released/eclipse/arduinoPlugin/packages/esp8266/hardware/esp8266/2.6.1/variants/nodemcu" -I"/home/erdem/eclipse/ide-latest-released/eclipse/arduinoPlugin/libraries/simplot/1.0.2" -MMD -MP -MF"core/core/uart.cpp.d" -MT"core/core/uart.cpp.o" -D__IN_ECLIPSE__=1 -x c++ "/home/erdem/eclipse/ide-latest-released/eclipse/arduinoPlugin/packages/esp8266/hardware/esp8266/2.6.1/cores/esp8266/uart.cpp"  -o  "core/core/uart.cpp.o"
In file included from /home/erdem/eclipse/ide-latest-released/eclipse/arduinoPlugin/packages/esp8266/hardware/esp8266/2.6.1/cores/esp8266/Arduino.h:286:0,
                 from /home/erdem/eclipse/ide-latest-released/eclipse/arduinoPlugin/packages/esp8266/hardware/esp8266/2.6.1/cores/esp8266/uart.cpp:43:
/home/erdem/eclipse/ide-latest-released/eclipse/arduinoPlugin/packages/esp8266/hardware/esp8266/2.6.1/variants/nodemcu/pins_arduino.h:36:0: warning: "LED_BUILTIN" redefined [enabled by default]
 #define LED_BUILTIN 2
 ^
<command-line>:0:0: note: this is the location of the previous definition
Finished building: /home/erdem/eclipse/ide-latest-released/eclipse/arduinoPlugin/packages/esp8266/hardware/esp8266/2.6.1/cores/esp8266/uart.cpp
...
```
I'm also opening a pull request for this issue.


### [MCVE](https://stackoverflow.com/help/mcve) Sketch

```cpp

#include <Arduino.h>

const byte pinLED = LED_BUILTIN;

void setup() {
	pinMode(pinLED, OUTPUT);
}

void loop() {
	digitalWrite(pinLED, HIGH);
	delay(1000);
	digitalWrite(pinLED, LOW);
	delay(1000);
}

```

### Debug Messages

--- a/variants/nodemcu/pins_arduino.h
+++ b/variants/nodemcu/pins_arduino.h
@@ -32,7 +32,9 @@
 static const uint8_t SDA = PIN_WIRE_SDA;
 static const uint8_t SCL = PIN_WIRE_SCL;
 
+#ifndef LED_BUILTIN
 #define LED_BUILTIN 2
+#endif
 #define LED_BUILTIN_AUX 16
 
 static const uint8_t D0   = 16;

